### ISSUE TITLE
Bug in save_model function

### ISSUE THREAD

**Issue by phil0042** [ebagdasa/backdoors101#25]
Hi,
the save_model function does not properly save the best checkpoint. The reason being the following two lines of code.
https://github.com/ebagdasa/backdoors101/blob/70869e5030eb282bb40161c58aa2f0fdf165d73b/helper.py#L50
https://github.com/ebagdasa/backdoors101/blob/70869e5030eb282bb40161c58aa2f0fdf165d73b/helper.py#L138
During training, save_model is called and loss_val contains the **accuracy** of the current iteration on the test set, not the loss value.

Fix:
Change the initial value of self.best_loss and modify the comparison (maybe rename self.best_loss and val_loss as well).
`self.best_loss = float(0)` and `if val_loss >= self.best_loss:`

**Comment by ebagdasa** [comment]
yeah, makes sense, please feel free to do a PR! Thanks for noticing it!
